cmd/codegen/generator/client_set.go

```diff
@@ -32,8 +32,8 @@ func InstallCodeGenerators(version string, gopath string) error {
 	}
 	for _, generator := range allGenerators {
 		path := fmt.Sprintf("%s/%s", basePath, generator)
-		util.AppLogger().Infof("installing %s in version %s via 'go get'", path, version)
-		err := util.GoGet(path, version, gopath, true, false)
+		util.AppLogger().Infof("installing %s version %s into %s", path, version, gopath)
+		err := util.GoGet(path, version, gopath, false, false)
```

Comment:
Hmm, I don't think this is going to work, is it? With Go modules enabled, I don't think to specify a version works. I guess that is why the generated sources differ now.

Reply:
Right, with Go modules enabled choosing a version is a bit dodgy due to the way go implemented modues - it does all sorts of weird stuff including mucking with your `go.mod`. If you use `go get` without go modules, the code will do the `go get -d` (i.e. download only), pause, adjust the committish in use manually using git, and then run `go install` to complete the `go get`.

In essence `go get` is not really suitable for use in modules mode in the same way as it was before modules - this is quite well documented in the go issue tracker.

Reply:
there is `gobin` which bypasses these issues and is probably the way to go for now until the go maintainers introduce better tooling.

Reply:
>  If you use go get without go modules, the code will do the go get -d (i.e. download only), pause, adjust the committish in use manually using git, and then run go install to complete the go get.

So, you  are saying it does honor versions, just in an "awkward" way?